File: 0901-1000/0905-Sort Array By Parity/0905-Sort Array By Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        int start = 0, end = A.size() - 1;
        while (start < end) {
            if (A[start] & 1) {
                swap(A[start], A[end--]);
            }
            else {
                start += 1;
            }
        }
        return  A;
    }
};

File: 0901-1000/0905-Sort Array By Parity/0905-Sort Array By Parity.py
class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        start = 0
        end = len(A) - 1
        while start < end:
            if A[start] & 1:
                A[start], A[end] = A[end], A[start]
                end -= 1
            else:
                start += 1
        return A
